posthog/queries/person_query.py

```diff
@@ -118,28 +120,17 @@ def get_query(
             f"""
             SELECT {fields}
             FROM person
+            {cohort_query if not person_filters else ""}
             WHERE team_id = %(team_id)s
-            {cohort_subquery}
+            {cohort_subquery if person_filters else ""}
```

Comment:
Now I didn't actually dig into why we have `cohort_query` vs. `cohort_subquery`, but that seemed to be the only difference between the `person_filters` and `not person_filters` versions of the query, so I consolidated this as seen here

Reply:
it's doing the same optimisation you're doing, but for cohorts

Reply:
... I think the reliance on `person_filters` here was because when both exist, the inner one can filter more aggressively

Reply:
Now that I think about it, even the search clause can go into that single subquery alongside person filters